src/beamlime/core/config_subscriber.py

```diff
@@ -32,22 +33,25 @@ def start(self):
         self._thread = threading.Thread(target=self._run_loop)
         self._thread.start()
 
+    def _process_message(self, message: KafkaMessage | None) -> None:
+        if message is None:
+            return
+        if message.error():
+            self._logger.error('Consumer error: %s', message.error())
+            return
+        key = message.key().decode('utf-8')
+        value = json.loads(message.value().decode('utf-8'))
+        self._logger.info(
+            'Updating config: %s = %s at %s', key, value, message.timestamp()
+        )
+        self._config[key] = value
```

Comment:
I think we should be careful of using `key` in this case.
`key` is for associating the messages with a certain partition, so it's mainly for the broker not the consumer.
https://www.confluent.io/learn/kafka-message-key/

Reply:
Moved to new issue #286, since this is unrelated to this change (just moved code).